astropy/io/votable/converters.py

```diff
@@ -674,19 +674,25 @@ def __init__(self, field, config=None, pos=None):
 
         precision = field.precision
         width = field.width
-        format_parts = ['%']
+
+        if precision is None:
+            format_parts = ['{0']
```

Comment:
Yes, this was the other most problematic part generating all kind of failures, and this was the hacky workaround I found to get rid of those failures. ``{0!r}`` definitely works, but it won't work as ``{06!r}`` and that case is definitely not covered with the tests either. What do you suggest, is there a case when width could be given, but not precision?